pkg/plugins/runtime/universal/outbound/outbound.go

```diff
@@ -77,7 +91,10 @@ func (v *VIPOutboundsReconciler) UpdateVIPOutbounds(ctx context.Context) error {
 			if dp.Spec.Networking.GetTransparentProxying() == nil {
 				continue
 			}
-			newOutbounds := dns.VIPOutbounds(dp.Meta.GetName(), dpList.Items, v.resolver.GetVIPs(), externalServices.Items)
+			allDps := make([]*mesh.DataplaneResource, len(ingresses)+len(dpList.Items))
```

Comment:
It's better to move this code out of the `for` loop because it doesn't depend on `dp` variable, so no point to execute it every time in a loop:
```go
allDps := make([]*mesh.DataplaneResource, len(ingresses)+len(dpList.Items))
copy(allDps[:len(ingresses)], ingresses)
copy(allDps[len(ingresses):], dpList.Items)

for _, dp := range dpList.Items {
    ...
    newOutbounds := dns.VIPOutbounds(model.MetaToResourceKey(dp.Meta), allDps, v.resolver.GetVIPs(), externalServices.Items)
    ...
}

Reply:
I'll do this in a sec